hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -381,7 +395,15 @@ private void handleUserTransaction(
             logger.error("An unexpected exception was thrown during handle", e);
             rollback(ResponseCodeEnum.FAIL_INVALID, stack, recordListBuilder);
             if (payer != null && fees != null) {
-                feeAccumulator.charge(payer, fees);
+                try {
+                    feeAccumulator.charge(payer, fees);
+                } catch (HandleException chargeException) {
+                    logger.error(
+                            "Unable to charge account {} a penalty after an unexpected exception {}. Cause of the failed charge:",
+                            payer,
```

Comment:
same here

Reply:
What does stop us from logging and throwing it?

Reply:
As written above, an `Exception` would skip creating and storing the `TransactionReceipt`, i.e., a user would not be able to see that the transaction failed.